homeassistant/components/__init__.py

```diff
@@ -27,6 +32,9 @@
 SERVICE_RELOAD_CORE_CONFIG = 'reload_core_config'
 SERVICE_CHECK_CONFIG = 'check_config'
 
+# How long between periodically saving the current states to disk
+STATE_DUMP_INTERVAL = timedelta(hours=1)
```

Comment:
Yeah, we would destroy any SD card faster if we store the state every time to disk. I think it's difficult to find a good value. But I think a database could be the job better as we can do with json.

I.e. my kids go to bed and I restart 30min later the system. Any room will restore the wrong state and all motion detection turn the light back to day lightning.

Reply:
Writing a file to disk every hour shouldn't be too bad. Not compared to what the recorder does today.

This is also for state restoration which is only for things like automations or devices with assumed state. This will in no way send commands to devices…

Reply:
Maybe all 15min?

A crash on wrong time with lost last state have horrible side effects on my full automated home.